test/unit/deferred.js

```diff
@@ -580,14 +582,17 @@ QUnit[ window.console ? "test" : "skip" ]( "jQuery.Deferred.exceptionHook", func
 			// Should NOT get an error
 			throw new Error( "Make me a sandwich" );
 		} ).then( null, jQuery.noop )
-	).then( function( ) {
+	).then( function barf( ) {
+		jQuery.thisDiesToo();
+	} ).then( null, function( ) {
 		window.console.warn = oldWarn;
 		done();
 	} );
 
 	defer.resolve();
 } );
 
+// Test fails in IE9 but is skipped there because console is not active
```

Comment:
Could you add a check for `(new Error()).stack` here so that the tests work even with IE9 DevTools enabled?